homeassistant/components/seventeentrack/coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""Coordinator for 17Track."""
+
+from dataclasses import dataclass
+from typing import Any
+
+from py17track import Client as SeventeenTrackClient
+from py17track.errors import SeventeenTrackError
+from py17track.package import Package
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import (
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    LOGGER,
+)
+
+
+@dataclass
+class SeventeenTrackData:
+    """Class for handling the data retrieval."""
+
+    def __init__(self) -> None:
+        """Initialize the data object."""
+        self.summary: dict[str, dict[str, Any]] = {}
+        self.live_packages: dict[str, Package] = {}
+        self.current_packages: set[Package] = set()
+
+
+class SeventeenTrackCoordinator(DataUpdateCoordinator[SeventeenTrackData]):
+    """Class to manage fetching 17Track data."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, client: SeventeenTrackClient) -> None:
+        """Initialize."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=DEFAULT_SCAN_INTERVAL,
+        )
+        self.show_delivered = self.config_entry.options[CONF_SHOW_DELIVERED]
+        self.account_id = client.profile.account_id
+        self.data = SeventeenTrackData()
```

Comment:
Why don't you construct a new one every time?

Reply:
fixed